src/logger.service.ts

```diff
@@ -27,77 +25,101 @@ const Levels = [
     this._log(NgxLoggerLevel.DEBUG, true, message, additional);
   }
 
-  info(message, ...additional: any[]) {
+  public info(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.INFO, true, message, additional);
   }
 
-  log(message, ...additional: any[]) {
+  public log(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.LOG, true, message, additional);
   }
 
-  warn(message, ...additional: any[]) {
+  public warn(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.WARN, true, message, additional);
   }
 
-  error(message, ...additional: any[]) {
+  public error(message, ...additional: any[]): void {
     this._log(NgxLoggerLevel.ERROR, true, message, additional);
   }
 
-  private _timestamp() {
-    return new Date().toISOString();
+  private _timestamp(): string {
+    return new Date()
+      .toISOString();
   }
 
-  private _logOnServer(level: NgxLoggerLevel, message, additional: any[]) {
-    if (!this.options.serverLoggingUrl) {
+  private _logOnServer(level: NgxLoggerLevel, message, additional: any[] = []): void {
+    // If the loggingUrl is not set or if the user provides a serverLogLevel and the current level is than that, do not log.
+    if (!this.options.serverLoggingUrl || level < this._serverLogLevel) {
       return;
     }
 
-    // if the user provides a serverLogLevel and the current level is than that do not log
-    if (level < this._serverLogLevel) {
-      return;
+    const headers = new HttpHeaders()
+      .set('Content-Type', 'application/json');
+
+    let messageToLog = '';
+
+    try {
+      messageToLog = JSON.stringify(message);
+    } catch (e) {
+      messageToLog = 'The provided "message" value could not be parsed with JSON.stringify().';
     }
 
-    const headers = new HttpHeaders().set('Content-Type', 'application/json');
 
-    this.http.post(this.options.serverLoggingUrl, {
-      level: Levels[level],
-      message: message,
-      additional: additional,
-      timestamp: this._timestamp()
-    }, {headers})
+    const additionalToLog: Array<string | null | undefined> = additional === null || additional === undefined ? []
+      : additional.map((val: any, idx: number) => {
+        try {
+          return val === null || val === undefined ? val
+            : typeof val === 'string' ? val
+              : JSON.stringify(val, null, 2);
+        } catch (e) {
+          return `The additional[${idx}] value could not be parsed using JSON.stringify().`;
+        }
+      });
+
+    this.http.post(this.options.serverLoggingUrl,
+      {
+        level: Levels[level],
+        message: messageToLog,
+        additional: additionalToLog,
+        timestamp: this._timestamp()
+      },
+      {
+        headers
+      })
       .subscribe(
-        res => null,
-        error => this._log(NgxLoggerLevel.ERROR, false, 'FAILED TO LOG ON SERVER')
+        (res: any) => this._log(NgxLoggerLevel.INFO, false, 'Server logging successful', [res]),
```

Comment:
lets changes this to TRACE. I don't know that I want my logs filling up with Server logging successful, every time I want to debug with level INFO.